runtime/shared_common/shrinit.cpp

```diff
@@ -1185,6 +1189,55 @@ j9shr_dump_help(J9JavaVM* vm, UDATA more)
 	j9file_printf(PORTLIB, J9PORT_TTY_OUT, "\n\n");
 }
 
+/**
+ * Notify the open/close state of jar/zip files so as to force a timestamp check.
+ *
+ * @param [in] vm The current J9JavaVM
+ * @param [in] classPathEntries  A pointer to the J9ClassPathEntry
+ * @param [in] entryCount  The count of entries on the classpath
+ * @param [in] isOpen  A flag indicating the open state for jar/zip files
+ */
+void
+j9shr_updateClasspathOpenState(J9JavaVM* vm, J9ClassPathEntry* classPathEntries, UDATA entryCount, BOOLEAN isOpen)
+{
+	J9SharedClassConfig* sharedClassConfig = vm->sharedClassConfig;
+	J9VMThread* currentThread = vm->internalVMFunctions->currentVMThread(vm);
+	UDATA newState = ((isOpen)? J9ZIP_STATE_OPEN : J9ZIP_STATE_CLOSED);
+	UDATA i = 0;
+
+	Trc_SHR_INIT_updateClasspathOpenState_entry(currentThread);
+
+	for (i = 0; i< entryCount; i++) {
+		U_16 cpeType = classPathEntries[i].type;
+		char* path = (char*) classPathEntries[i].path;
+		IDATA pathLen = classPathEntries[i].pathLength;
+		bool isJarZipFile = false;
+
+		switch(cpeType) {
+		case CPE_TYPE_JAR:
+			isJarZipFile = true;
+			break;
+		case CPE_TYPE_UNKNOWN:
+			if (pathLen > 4) {
+				char* endsWith = path + (pathLen - 4);
+				if ((0 == strcmp(endsWith, ".jar"))
```

Comment:
Is it necessary to check this here? getCpeTypeForProtocol() already does this.

Reply:
Agree and already removed here.